src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesQueueTaskDispatcher.java

```diff
@@ -0,0 +1,62 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+import hudson.Extension;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import hudson.model.Node;
+import hudson.model.Queue;
+import hudson.model.Queue.Task;
+
+import hudson.model.queue.CauseOfBlockage;
+import hudson.model.queue.QueueTaskDispatcher;
+import jenkins.model.Jenkins;
+
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Extension
+@SuppressWarnings({"rawtypes"})
+public class KubernetesQueueTaskDispatcher extends QueueTaskDispatcher {
+
+    @Override
+    public CauseOfBlockage canTake(Node node, Queue.Task item) {
+        if (node instanceof KubernetesSlave) {
+            KubernetesSlave slave = (KubernetesSlave) node;
+            Task ownerTask = item.getOwnerTask();
+            if (ownerTask instanceof WorkflowJob) {
+                WorkflowJob workflowJob = (WorkflowJob) ownerTask;
+                return check(slave, workflowJob);
+            }
+            if (item instanceof Job) {
+                return check(slave, (Job) item);
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public CauseOfBlockage canTake(Node node, Queue.BuildableItem item) {
+        return this.canTake(node, item.task);
+    }
+
+    private CauseOfBlockage check(KubernetesSlave agent, Job job) {
```

Comment:
Would work better if this was moved to `KubernetesFolderProperty` with something like `static boolean isAllowed(KubernetesSlave agent, Job job);`

Reply:
Good idea! Fixed